### ISSUE TITLE
Cast from complex to real fails on CUDA and OpenCL backends

### ISSUE THREAD

**Issue by shehzan10** [arrayfire/arrayfire#1648]
Sample Code:
```
array ac32 = randu(5, c32);
array ac64 = randu(5, c64);
array bf32 = ac32.as(f32); // Fails
array bf64 = ac32.as(f64); // Fails
array cf32 = ac64.as(f32); // Fails
array cf64 = ac64.as(f64); // Fails
```

Does not fail on CPU backend.

**Comment by pavanky** [comment]
Hmm this should be an exception and may be a message asking the user to explicitly use `abs`, `real` or `img`.

**Comment by shehzan10** [comment]
I believe right now it (at least CPU) is doing `abs` internally.

**Comment by shehzan10** [comment]
Fixed in hotfix-3.4.2
